src/lib.rs

```diff
@@ -94,59 +113,129 @@ fn write_mod_cmd<W: Write>(config: &ModConfig, ctx: &ConfigContext, mut writer:
+
     for task in &config.tasks {
-        match task {
-            Task::InvokeScc {
-                path,
-                custom_cache_dir,
-                terminate_on_errors,
-            } => {
-                let cmd = Path::new(ctx.game_dir)
-                    .join("engine")
-                    .join("tools")
-                    .join("scc.exe");
-                let path = render(path, ctx.clone());
-                let custom_bundle = render(custom_cache_dir, ctx.clone());
-                const NO_WINDOW_FLAGS: u32 = 0x08000000;
-
-                let res = Command::new(&cmd)
-                    .arg("-compile")
-                    .arg(path)
-                    .arg("-customCacheDir")
-                    .arg(custom_bundle)
-                    .creation_flags(NO_WINDOW_FLAGS)
-                    .status()
-                    .ok();
-                if *terminate_on_errors == Some(true) && !matches!(res, Some(st) if st.success()) {
-                    std::process::exit(0);
+        let mut tasks_done = TASKS_DONE.lock().unwrap();
+        if tasks_done.insert(task.command.clone()) {
+            log.debug(format!("Running task: \"{}\"", task.command));
+
+            match task.command.as_str() {
+                "InvokeScc" => {
+                    let cmd = Path::new(ctx.game_dir)
+                        .join("engine")
+                        .join("tools")
+                        .join("scc.exe");
+                    let path = render(task.path.as_str(), ctx.clone());
+                    let custom_bundle = render(task.custom_cache_dir.as_str(), ctx.clone());
+
+                    let res = Command::new(&cmd)
+                        .arg("-compile")
+                        .arg(path)
+                        .arg("-customCacheDir")
+                        .arg(custom_bundle)
+                        .creation_flags(NO_WINDOW_FLAGS)
+                        .status()
+                        .ok();
+                    if task.terminate_on_errors && !matches!(res, Some(st) if st.success()) {
+                        std::process::exit(0);
+                    }
+                }
+                _ => {
+                    let cmd_path = Path::new(cybercmd_path)
+                        .join(format!("{}.exe", task.command));
+                    let cmd = cmd_path.canonicalize();
+
+                    if let Ok(cmd) = cmd {
+                        if cmd.starts_with(cybercmd_path) {
+                            let res = Command::new(&cmd)
+                                .arg("-cybercmd")
+                                .current_dir(ctx.game_dir)
+                                .creation_flags(NO_WINDOW_FLAGS)
+                                .status()
+                                .ok();
+                            if !matches!(res, Some(st) if st.success()) {
+                                log.error(format!("Task \"{}\" failed when run.", task.command));
```

Comment:
this logger is a bit awkward to use, normally a logger should accept `impl Display` rather than a `String` (it could then be wrapped with a macro that forwards `format_args!`)
but rather than writing a custom logger, I'd check for a small crate that can do file logging for us, I think there should be something

Reply:
There are a few. I started with one of the premade loggers. I couldn't get it to work properly. So I wrote my own custom one to learn some more Rust.

Now in hindsight I think the logger was getting initialized twice, and that resulted in the file being overwritten.

I definitely have more enthusiasm than knowledge when it comes to Rust and updating cybercmd. 😂

Reply:
I think there's an ordering issue between our entry point, and the initialization of the static rust logger, which screws up the default logger most libraries use
There might still be crates that expose a logger without having to use the static

Reply:
I'll make a few comments on the logger, I think it can do with some minor adjustments